engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/io/PrintlnNode.java

```diff
@@ -91,8 +93,12 @@ private String fallbackToString(Object obj) {
   }
 
   @CompilerDirectives.TruffleBoundary
-  private void print(PrintStream out, Object str) {
-    out.println(str);
+  private void print(PrintStream out, Object str, String nl) {
+    switch (nl) {
+      case "\n" -> out.println(str);
+      case "" -> out.print(str);
+      default -> out.print(str + nl);
+    }
```

Comment:
At this point, wouldn't it be easier to just have 2 separate nodes - one for `print` and one for `println`?

Each node would have much simpler logic then.

Reply:
The less builtins the better, but I can replace the second argument with `boolean`.

Reply:
On the other hand. The builtin is in `private` module. As such it is not part of accessible API and it can be changed any time...